mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +46,239 @@
     load_index,
     resolve_es_config_for_version,
 )
+from ._util import _is_module_available, _get_module_or_attr
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
 
 logger = logging.getLogger(__name__)
 
 
 class EntityResolver:
-    """An entity resolver is used to resolve entities in a given query to their canonical values
-    (usually linked to specific entries in a knowledge base).
+
+    @staticmethod
+    def validate_resolver_name(name):
+        if name not in ENTITY_RESOLVER_MODEL_TYPES:
+            msg = "Expected 'model_type' in ENTITY_RESOLVER_CONFIG among {!r}"
+            raise Exception(msg.format(ENTITY_RESOLVER_MODEL_TYPES))
+        if name == "sbert_cosine_similarity" and not _is_module_available("sentence_transformers"):
+            raise ImportError(
+                "Must install the extra [bert] to use the built in embbedder for entity "
+                "resolution. See https://www.mindmeld.com/docs/userguide/getting_started.html")
+
+    def get_resolver(self, app_path, resource_loader, entity_type, **kwargs):
+        """
+        Identifies appropriate entity resolver based on input config and
+            returns it.
+
+        Args:
+            app_path (str): The application path.
+            resource_loader (ResourceLoader): An object which can load resources for the resolver.
+            entity_type (str): The entity type associated with this entity resolver.
+            er_config (dict): A classifier config
+            es_host (str): The Elasticsearch host server.
+            es_client (Elasticsearch): The Elasticsearch client.
+        """
+        er_config = (
+            kwargs.pop("er_config", None) or
+            get_classifier_config("entity_resolution", app_path=app_path)
+        )
+        resolver_name = er_config.get("model_type", None)
+        self.validate_resolver_name(resolver_name)
+        return ENTITY_RESOLVER_MODEL_MAPPINGS.get(resolver_name)(
+            app_path, resource_loader, entity_type, er_config, **kwargs
+        )
+
+
+class EntityResolverBase(ABC):
```

Comment:
not sure if it's a better name but i wonder if it could be called `EntityResolverModelBase` to somewhat distinguish it from the `EntityResolver` class?